src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -37,17 +36,19 @@ internal struct ReadStackFrame
         public bool Drain;
 
         public bool IsDictionary => JsonClassInfo.ClassType == ClassType.Dictionary;
+        public bool IsDictionaryProperty => JsonPropertyInfo != null ? !JsonPropertyInfo.IsPropertyPolicy && JsonPropertyInfo.ClassType == ClassType.Dictionary : false;
```

Comment:
nit: Can we expand this to multi-line getter format? It's difficult to parse (or a the very least add line breaks).